util/progress/progressui/json.go

```diff
@@ -0,0 +1,38 @@
+package progressui
+
+import (
+	"time"
+
+	"github.com/moby/buildkit/client"
+	"github.com/moby/buildkit/solver/pb"
+)
+
+type Vertex struct {
+	*client.Vertex
+	Statuses []VertexStatus         `json:"statuses,omitempty"`
+	Logs     map[string][]VertexLog `json:"logs,omitempty"`
+	Warnings []VertexWarning        `json:"warnings,omitempty"`
+}
+
+type VertexStatus struct {
+	ID        string     `json:"id"`
+	Name      string     `json:"name,omitempty"`
+	Total     int64      `json:"total,omitempty"`
+	Started   *time.Time `json:"started,omitempty"`
+	Completed *time.Time `json:"completed,omitempty"`
+}
+
+type VertexLog struct {
+	Timestamp float64 `json:"ts"`
+	Message   string  `json:"msg"`
+}
+
+type VertexWarning struct {
+	Level  int      `json:"level,omitempty"`
+	Short  string   `json:"short,omitempty"`
+	Detail []string `json:"detail,omitempty"`
+	URL    string   `json:"url,omitempty"`
+
+	SourceInfo *pb.SourceInfo `json:"sourceInfo,omitempty"`
```

Comment:
I guess we don't want to have this `pb` info in the output?

Reply:
I can remove this. I figured the source info might be useful and the protobuf structure is already annotated with the correct json markers.